upi/vsphere/nsx_network/main.tf

```diff
@@ -0,0 +1,67 @@
+resource "nsxt_logical_port" "logical_port" {
+  admin_state       = "UP"
+  description       = "LP1 provisioned by Terraform"
+  display_name      = "LP1"
+  logical_switch_id = "${var.logical_switch}"
+}
+
+resource "nsxt_logical_router_downlink_port" "downlink_port" {
+  description                   = "DP1 provisioned by Terraform"
+  display_name                  = "DP1"
+  logical_router_id             = "${var.t1_router}"
```

Comment:
it seems a bit weird that we don't create router and switch as well. What is the reason?

Reply:
Currently, the reason for that is the fact that the DNS that serves the installation and the cluster as to be in a network that is accessible to the VMs. therefore, the logical switch and t1 router need to be created in advance. This needs to be changed eventually.

Reply:
added the proper elements up to t0 router.